### DISCUSSION TITLE
[11 settembre 2024] Automi

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/automi-calcolabilita-complessita#34] [Automi]
![image](https://github.com/user-attachments/assets/bfff3c3b-5070-43b1-bc4c-0756cf50b072)

**Comment by cornflexxx** [comment]
Consideriamo la $NPDA$  $P'$ che simula l'esecuzione della $DFA$ $D$ che riconosce $R$ e del $NPDA$ $P$ che riconosce $C$.
$P'$  prende una stringa $w\in \Sigma^*$ e tiene traccia dell'esecuzione sia di $D$ che di $P$ su $w$, quindi **accetta** la stringa solo se sia $P$ che $D$ la **accettano**. 

-----------------
Assumiamo che  $L = \\{w\in \\{a,b,c\\}^*t.c. |w_a|   =| w_b |= |w_c|\\}$ sia in $CFL$, e  $R \in REG$ . Per la proposizione precedente $L\cap R = Q$ è  _**context free**_ , ma $Q =\\{a^n b^n c^n | n\ge 0\\}$ è ben noto non essere un linguaggio  _**context free**_ , quindi o $L\notin CFL$.

(sia $R = a^* b^* c^*$)
